server/controllers/UserController.js

```diff
@@ -0,0 +1,94 @@
+import models from '../database/models';
+import helpers from '../helpers';
+
+const { FavoriteBook, Book, Author } = models;
+const { responseMessage } = helpers;
+
+/**
+ * @class UserController
+ * @classdesc controller for all user functionality
+ */
+export default class UserController {
+  /**
+   * @description check if book is in database
+   * @param {object} request express request object
+   * @returns {array} favoritebook, book and param object in an array
+   * @memberof UserController
+   */
+  static async findBook(request) {
+    const { id } = request.userData;
+    const { bookId } = request.params;
+    const favoritebook = await FavoriteBook.findOne({ where: { userId: id, bookId } });
+    const book = await Book.findOne({ where: { id: bookId }, include: [{ model: Author }] });
+    return [favoritebook, book, { userId: id, bookId: parseInt(bookId, 10) }];
+  }
+
+  /**
+   * @description user can favorite a book
+   * @param {object} request express request object
+   * @param {object} response express response object
+   * @returns {json} json
+   * @memberof UserController
+   */
+  static async favoriteBook(request, response) {
+    try {
+      const [favoritebook, book, param] = await UserController.findBook(request);
+      if (!book) return responseMessage(response, 404, { message: 'book not found' });
+      if (favoritebook) return responseMessage(response, 409, { message: 'book already in your favourites' });
+      const favorite = await FavoriteBook.create(param);
+
+      if (favorite) {
+        return responseMessage(response, 200, {
```

Comment:
201 statusCode is recommended here since a new instance of FavouriteBook model was created.

Reply:
I believed 200 was okay since it's being added/removed to/from favorite list. Will change it

Reply:
done